src/FilenameParsing/HashFileIDHelper.php

```diff
@@ -14,70 +13,13 @@
  *
  * e.g.: `Uploads/a1312bc34d/sam__ResizedImageWzYwLDgwXQ.jpg`
  */
-class HashFileIDHelper implements FileIDHelper
+class HashFileIDHelper extends AbstractFileIDHelper
```

Comment:
This is BC safe - anything checking for types of `FileIDHelper` will still pass because `AbstractFileIDHelper` implements it.

This is similar to when we added an `AbstractGridfieldComponent` class.